core/include/thread.h

```diff
@@ -398,6 +398,22 @@ void thread_yield(void);
  */
 void thread_yield_higher(void);
 
+/**
+ * @brief   Puts the current thread into zombie mode. Has to be terminated externally.
```

Comment:
Feel free to elaborate a bit on what that means.
Something along the line:

```
A thread in zombie state will never be scheduled again, but its scheduler entry and stack will be kept.
It is supposed to be cleaned up by @ref thread_kill_zombie().
```

Reply:
Yes, the thread becomes a zombie thread (will never be  scheduled again, but its scheduler entry and stack will be kept ).
And is supposed to be cleaned up by @ref thread_kill_zombie() (run from different thread/not the zombie thread because the zombie should never be able to run again)